contracts/Voter.sol

```diff
@@ -386,4 +414,23 @@ contract Voter is IVoter, ERC2771Context, ReentrancyGuard {
       _distribute(_gauges[x]);
     }
   }
+
+  function depositLP(address _pool, uint256 _amount) external hasValidGauge(_pool) nonReentrant {
+    if (_amount == 0) revert ZeroAmount();
+    address _sender = msg.sender;
+    address _gauge = gauges[_pool];
+    IERC20(_pool).safeTransferFrom(_sender, address(this), _amount);
+    IERC20(_pool).approve(_gauge, _amount);
+    IGauge(_gauge).deposit(_amount, _sender);
```

Comment:
users are able to deposit to a gauge directly, as a result, the incentive won't be updated. Meanwhile, gaugeToIncentives is the only thing needed. Thus, could we move incentive of a gauge into the gauge, such that when there is a deposit to the gauge, we can update its corresponding incentive.

Reply:
Gauge and incentive are associated with each other and callback is called in gauge when deposit.